plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java

```diff
@@ -28,17 +28,26 @@
 import java.nio.file.Path;
 
 /**
- * Utililty to find the Google Cloud SDK either at locations configured by the user or in standard
+ * Utility to find the Google Cloud SDK either at locations configured by the user or in standard
  * locations on the system.
  */
 public class CloudSdkProvider extends ContextFunction {
 
+  private final IPreferenceStore preferences;
+
+   public CloudSdkProvider(IPreferenceStore preferences) {
```

Comment:
or we remove this and pass in actual non-null preferences from everywhere. Haven't yet decided. May depend on the dependencies that would introduce into other plug-ins.